catalog/app/containers/Search/model.tsx

```diff
@@ -0,0 +1,342 @@
+import invariant from 'invariant'
+import * as R from 'ramda'
+import * as React from 'react'
+import * as RR from 'react-router-dom'
+
+import * as Model from 'model'
+import * as GQL from 'utils/GraphQL'
+// import * as NamedRoutes from 'utils/NamedRoutes'
+import parseSearch from 'utils/parseSearch'
+import useMemoEq from 'utils/useMemoEq'
+
+import BASE_SEARCH_QUERY from './gql/BaseSearch.generated'
+// import FACET_QUERY from './gql/Facet.generated'
+import FIRST_PAGE_QUERY from './gql/FirstPage.generated'
+import NEXT_PAGE_QUERY from './gql/NextPage.generated'
+
+function ifChanged<T>(newValue: T) {
+  return (oldValue: T) => (R.equals(newValue, oldValue) ? oldValue : newValue)
+}
+
+enum ResultType {
+  Objects = 'objects',
+  Packages = 'packages',
+  Any = 'any',
+}
+
+interface ResultTypeFacetState {
+  selected: ResultType
```

Comment:
Good old `value` is good enough. Also, it can be confusing to read `selected` not as `boolean`

Reply:
yeah ok